src/plays/simple-live-chat/components/chat/chat-body.jsx

```diff
@@ -0,0 +1,132 @@
+import format from "date-fns/format";
+import { Fragment, useEffect, useRef, useState } from "react";
+
+// css
+import "./chat-body.scss";
+
+// assets
+import userIcon from "../../images/user_icon.png";
+
+const User = ({ photoURL, onClick }) => {
+  const [defaultImage, setDefaultImage] = useState(photoURL);
+  return (
+    <img
+      src={defaultImage}
+      alt='avatar'
+      onClick={onClick}
+      className='simple-live-chat-user'
+      onError={() => setDefaultImage(userIcon)}
+    />
+  );
+};
+
+const ChatBody = ({ message, uid, loading }) => {
+  const [node, setNode] = useState([]);
+  const scrollHandler = useRef();
+
+  const dateTimeFormatter = (val) => {
+    return format(new Date(val * 1000), "hh:mm bbb");
+  };
+
+  const showDateHandler = (array = [], currIndex) => {
+    const formattedDate = format(
+      new Date(array[currIndex]?.createdAt?.seconds * 1000),
+      "dd MMM yyyy"
+    );
+    if (currIndex === 0) return formattedDate;
+    const prev = array[currIndex - 1];
+    const prevDate = new Date(prev?.createdAt * 1000).getDate();
+    const currDate = new Date(array[currIndex].createdAt * 1000).getDate();
+
+    if (prevDate !== currDate) {
+      return formattedDate;
+    }
+    return null;
+  };
+
+  const DateDisplayComp = ({ idx, message }) => {
+    const date = showDateHandler(message, idx);
+    if (!date) return null;
+    return (
+      <div className='simple-live-chat-date-display'>
+        <p>{date}</p>
+      </div>
+    );
+  };
+
+  const displaySenderNameHandler = (val) => () => {
+    if (node.includes(val)) {
+      return setNode(node.filter((item) => item !== val));
+    }
+    return setNode([...node, val]);
+  };
+
+  useEffect(() => {
+    scrollHandler.current?.scrollIntoView();
+  }, [message]);
+
+  const DisplayNameComponent = ({ displayName, sender, id }) => {
+    if (!node.includes(id)) return null;
+    return (
+      <div className={sender ? "simple-live-chat-text-right" : "simple-live-chat-text-left"}>{displayName}</div>
```

Comment:
Minor: a more simplistic approach would be
```
<div className={`simple-live-chat-text-${sender? 'right : 'left'}`}>{displayName}</div>
```

Reply:
yes the following code added.